### DISCUSSION TITLE
Does k8up restore pvc snapshot    to destination cluster

### DISCUSSION THREAD

**Discussion by vfolk30** [k8up-io/k8up#804] [Q&A]
Hi,
I would like  to know k8up has functionality to restore pvc from one cluster to another cluster ?

**Comment by Kidswiss** [comment]
Hi @vfolk30 

You should be able to restore a backup from one cluster to another. You have to provide the right S3 credentials to the restore process, and it should work: https://k8up.io/k8up/2.5/how-tos/restore.html#_restore_from_s3_to_pvc

The example on the page will restore the latest snapshot it finds in the repository. If you need another snapshot you can specify `spec.snapshot`.

**Reply by vfolk30** [reply]
Thanks @Kidswiss  for helping here.

Could you also  explain me about " backup repository for its integrity." 
How it works?  what it check and how?
